include/ensmallen_bits/lookahead/lookahead_impl.hpp

```diff
@@ -188,7 +188,7 @@ Lookahead<BaseOptimizerType, DecayPolicyType>::Optimize(
     size_t batchSize = 1;
     // Check if the optimizer implements the BatchSize() method and use the
     // parameter for the objective calculation.
-    if (traits::HasBatchSizeSignature<BaseOptimizerType>::value)
+    if constexpr (traits::HasBatchSizeSignature<BaseOptimizerType>::value)
```

Comment:
`if constexpr` is C++17 onwards.  ensmallen requires only C++14.  I'd prefer to keep the standard at C++14 unless there is really a compelling reason to bump it.

In this case, `if constexpr` doesn't actually help to produce any more efficient machine compared to standard `if` expression.  Any optimising compiler (all the way back to C++98 days) will figure out that `traits::HasBatchSizeSignature<...>::value` is a compile time constant (even when declared via old-school `static const`) and optimise accordingly.